torchvision/transforms.py

```diff
@@ -236,13 +236,52 @@ class Pad(object):
         padding (int or sequence): Padding on each border. If a sequence of
             length 4, it is used to pad left, top, right and bottom borders respectively.
         fill: Pixel fill value. Default is 0.
+        type:padding type: constant,reflect,edge,symmetric
     """
 
-    def __init__(self, padding, fill=0):
+    def __init__(self, padding, fill=0,type="constant"):
         assert isinstance(padding, numbers.Number)
         assert isinstance(fill, numbers.Number) or isinstance(fill, str) or isinstance(fill, tuple)
         self.padding = padding
         self.fill = fill
+        assert (type in ["constant","edge","symmetric","reflect"])
+        self.type = type
+
+    def __expand_reflect(slef,image, border=0):
```

Comment:
I don't think you should use double _ `__` here as this is used for name mangling, these can just have a single `_` indicating private.